packages/vite/src/node/config.ts

```diff
@@ -607,6 +588,24 @@ export async function resolveConfig(
   const BASE_URL = resolvedBase
 
   // resolve worker
+  let workerConfig = mergeConfig({}, config)
```

Comment:
Is it correct that we resolve the worker plugins `config` hook using a `config` that has been mutated by `config.plugins`?

That means that `config.plugins` has an effect on the worker config, but they will not be used for `load` or `transform` for example since worker plugins are the only one that takes effect.

Reply:
I think we may need to improve the docs, but it makes sense to me that the main plugins can modify the full config. For example, maybe some regular plugins (like a SvelteKit plugin, may want to add workers plugins).
IMO the workers should see the final config, it doesn't matter if you use `vite.config.js` or a plugin to change an option.

Reply:
Thank for review. `resolvedConfig` must need to split effect. It make worker build plugin use the diff state. 

I am not very sure the worker config use the state changed by the plugin or the inline config?

Reply:
In the current PR it is using the modified config by the plugins, but I think that is how it should work. 

What do you mean by "resolvedConfig must need to split effect. It make worker build plugin use the diff state."?

Reply:
`resolvedConfig` must be a new object. Because build worker plugin should use the diff state from the weakmap.

And now call order is 
1. modify user config
2. deep clone user config for worker config
3. call worker plugin `config` hook

I am thinking about the `config` is really need to deep clone one? Should we allow the worker plugin modify the user config?

Reply:
No, I think worker plugins shouldn't modify the main config. The deep clone is ok here 👍🏼

Reply:
so we are allow the worker config use the state changed by the plugin. I will feat this point in the document

Reply:
I think we should move forward with the PR in its current state, and update the docs after discussing it in a team meeting, just in case
@bluwy are you ok with this? Or do you still have doubts about the current change set?